src/main/java/org/intellij/sdk/codesync/alerts/PricingAlerts.java

```diff
@@ -18,25 +18,41 @@ private static void acquirePricingLock() {
         CodeSyncLock pricingAlertLock = new CodeSyncLock(LockFileType.PROJECT_LOCK, PRICING_ALERT_LOCK_KEY);
         pricingAlertLock.acquireLock();
     }
+
     public static void setPlanLimitReached() {
         setPlanLimitReached((Project) null);
     }
+
     public static void setPlanLimitReached(Project project) {
         // We only want to show notification once every 5 minutes, I have implemented that using locks with an expiry of
         // 5 minutes. So, skip the notification if lock is not acquired.
         acquirePricingLock();
         CommonUtils.invokeAndWait(
-            () -> {
-                PricingAlertDialog pricingAlertDialog = new PricingAlertDialog(false, false, CODESYNC_PRICING_URL, project);
-                return pricingAlertDialog.showAndGet();
-            },
-            ModalityState.defaultModalityState()
+                () -> {
+                    PricingAlertDialog pricingAlertDialog = new PricingAlertDialog(false, false, CODESYNC_PRICING_URL, project);
```

Comment:
Do we know why are we passing hardcoded false for canAvailTrial here @saadberry ?

Reply:
I believe that is the default case when we cannot assess wether user is eligible for trial or not

Reply:
Alright, can you kindly open a ticket for this to use actual API call here since you have added that API call now

Reply:
Done: https://linear.app/codesync/issue/PLU-8/fetch-canavailtrial-from-subscription-api